src/main/java/org/apache/maven/shared/utils/cli/Commandline.java

```diff
@@ -241,15 +241,16 @@ public void addSystemEnvironment()
     public String[] getEnvironmentVariables()
     {
         addSystemEnvironment();
-        String[] environmentVars = new String[envVars.size()];
-        int i = 0;
+        List<String> environmentVars = new ArrayList<>();
         for ( String name : envVars.keySet() )
```

Comment:
While I think the solution is working, I wonder whether it solves the actual cause: At least on Windows, it is impossible to stored environment variables with a `NULL` value. Try this effectly removes it from the environment, according to the Windows API docs. So at least on Windows the correct solution would be that this variable name-value-pair must not be put into `envVars` right from the start. In turn, *here* is no change to be made.

Can someone conform that it is valid on other OS to actually have an environment variable in the environment that stores `NULL` (not the String but `null` in the Java sense)?

Reply:
reply on jira issue - here we don't set system environment it happen later

Reply:
I agree this is a real bug. Rather than fixing it here, perhaps `addEnvironmentVariable` should throw a NullPointerException for null values?

Reply:
Throwing exception in this case will has impact on caller, eg has impact on https://issues.apache.org/jira/browse/MINVOKER-273

In my proposition we have new feature - when environment variable has null value will be removed from executing job environment.

Look at unit tests.

Different approach can be  throw exception for null values and add next method to possibility remove/exclude specific environment variables.